performance/reports/tests/test_rp.py

```diff
@@ -71,3 +70,76 @@ def test_export_metrics_to_csv(mock_series_to_csv, mock_dataframe_to_csv, _):
         call(os.path.join(report_output_path, f'rp_report-{date_start}-rp2.csv'))
 
     ]
+
+
+def test_transform_metrics_calculates_computed_metrics_accurately():
+    df = pandas.DataFrame.from_dict({
+        0: ['RP 1', 200, 100, 150.0, 250.0, 50.0, 20.0, 30.0],
+        1: ['RP 2', 500, 200, 100.0, 200.0, 30.0, 10.0, 20.0]
+    },
+        orient="index",
+        columns=["rp", "signup_success", "signin_success",
+                 "signin_attempt", "signup_attempt", "single_idp_attempt",
+                 "visits_will_not_work", "visits_might_not_work"])
+    additional_metric_df = pandas.DataFrame.from_dict({
+        0: [450, 300, 200 / 300, 100 / 300],
+        1: [330.0, 700, 500 / 700, 200 / 700]
+    },
+        orient="index",
+        columns=["all_referrals_with_intent", "success", "success_fraction_signup",
+                 "success_fraction_signin"])
+    expected_df = df.copy(deep=True).join(additional_metric_df)
+
+    transform_metrics(df)
+
+    assert_frame_equal(df, expected_df)
+
+
+@patch.object(piwik, 'get_visits_might_not_work')
+@patch.object(piwik, 'get_visits_will_not_work')
+@patch.object(piwik, 'get_all_single_idp_attempts_for_rp')
+@patch.object(piwik, 'get_all_signup_attempts_for_rp')
+@patch.object(piwik, 'get_all_signin_attempts_for_rp')
+def test_add_piwik_data_for_rp(mock_get_all_signin_attempts_for_rp, mock_get_all_signup_attempts_for_rp,
```

Comment:
I'm not entirely convinced this actually tests anything interesting, but eh :-)

Reply:
The function `add_piwik_data_for_rp` invokes a few Piwik APIs to fetch data such as all signin attempts for an RP. This test ensures that the results returned by Piwik are correctly added to the DataFrame.